src/main/java/org/opensearch/security/support/Base64Helper.java

```diff
@@ -88,10 +95,40 @@ public class Base64Helper {
         Enum.class
     );
 
+
+    private enum CustomSerializationFormat {
+
+        WRITEABLE(1),
+        STREAMABLE(2),
+        GENERIC(3);
+
+        private final int id;
+
+        CustomSerializationFormat(int id) {
+            this.id = id;
+        }
+
+        static CustomSerializationFormat fromId(int id) {
+            switch (id) {
+                case 1: return WRITEABLE;
+                case 2: return STREAMABLE;
+                case 3: return GENERIC;
+                default: throw new IllegalArgumentException(String.format("%d is not a valid id", id));
+            }
+        }
+
+    }
+
+    private static final BiMap<Class<?>, Integer> writeableClassToIdMap = HashBiMap.create();
+    private static final StreamableRegistry streamableRegistry = StreamableRegistry.getInstance();
     private static final Set<String> SAFE_CLASS_NAMES = Collections.singleton(
         "org.ldaptive.LdapAttribute$LdapAttributeValues"
     );
 
+    static {
+        registerAllWriteables();
```

Comment:
why not register Streamables here?

Reply:
The logic for streamable registration is kept internal to the `StreambleRegistry` class as it needs to preserve registration ordering to avoid any breaking change in the type IDs that it internally assigns to the streamables. Further, for simplification, we do not expose any methods for registering streamables from outside `StreamableRegistry` class and all registrations are to reside within this class only.

We just need to get its instance via `StreamableRegistry::getInstance()` and invoke `writeTo()` and `readFrom()` methods.